.pre-commit-config.yaml

```
@@ -16,3 +16,11 @@ repos:
     rev: v0.10.0
     hooks:
       - id: shellcheck
+  - repo: local
+    hooks:
+      - id: check-api
+        name: check old API use
+        entry: tools/check-old-api.py
```

Maybe rename all these to include `precice`? It took me a bit to understand the context when I first opened this PR.

```suggestion
      - id: check-precice-api
        name: check old preCICE API use
        entry: tools/check-old-precice-api.py
```